willYouSleep2.0.js

```diff
@@ -0,0 +1,320 @@
+  console.clear();
+  console.log("You got down to the basement and the power is out \n");
+let person = readline1.question(
+    "A: Go back up to get flashlight?     B: Keep going and you'll find The LIGHT!! \n"
+  );
+person = person.toUpperCase()
+  if (person === "A") {
+    getFlashlight();
+  } else if (person === "B") {
+    keepGoing();
+  }
+}
+
+function getFlashlight() {
+  console.clear();
+  console.log("You on your way back upstairs \n");
+  console.log("You get to your room \n");
+  console.log("But your bed is calling you \n");
+let person = readline1.question("A: Go to sleep ?    B: Find flashlight? \n");
+person = person.toUpperCase()
+  if (person === "A") {
+    goToSleep();
+  } else if (person === "B") {
+    findFlashlight();
+  }
+}
+
+function keepGoing() {
+  console.clear();
+  console.log(
+    "You trying to move around the basement without killing yourself in the dark \n"
+  );
+  console.log(
+    "It just occurred to you that you're terribly scared of the dark!"
+  );
+  console.log("You hear a loud Noise!! \n");
+  person = readline1.keyInYN("Do You run out and SCREAM BLOODY MURDER!! \n");
+  if (person) {
+    console.clear();
+    console.log(
+      "Neighbor B laughs at you and ask if you want to press charges. \n"
+    );
+    console.log("Neighbor A scolds at you. " + '"I told you to STAY put!!" \n');
+    console.log(
+      "You roll your eyes and walk past them to your room and go to sleep! \n"
+    );
+  } else {
+    console.log(
+      "\nIf loud noises don't scare you! Why the heck did you wake up in the first place! GO BACK TO BED!! \n"
+    );
+  }
+
+  startGame();
+}
+
+function findFlashlight() {
+  console.clear();
+  console.log(
+    "You in your room and find the flashlight but it doesn't have any batteries \n"
+  );
+  console.log("You're tired!! You was woken out of your sleep. Do you... \n");
+let person = readline1.question(
+    "A: Look for batteries?         B: Who has time to look for batteries? \n"
+  );
+  person =  person.toUpperCase()
+  if (person === "A") {
+    batteries();
+  } else if (person === "B") {
+    goToSleep();
+  }
+}
+
+function batteries() {
+console.clear()
+let num = batteryNumber()
+battery = num * 3
+console.log(
+    `Actually do you really want to go off looking for ${battery} batteries? \n`
+  );
+  console.log(
+    `I didn't think so. by the time you do find ${battery} batteries it will be daylight! \n`
+  );
+  console.log(
+    "Your brain is not even capable enough right now to even do the math.  \n"
+  );
+let next = readline1.keyInYN(`Isn't ${battery} too many?! Aren't you ready to go to sleep yet?`)
+if(next){
+    goToSleep()
+} else {
+    storyTime1()
+    
+}
+}
+function storyTime1(){
+    let arr = ['pillow','duvet','cloud nine', 'sheets']
+    let str = `\nOnce upon a time, there was a...`
+    let array = []
+    for(let i = 0; i < arr.length; i++){
+    
+      array.push(arr[i])
+    }
+    let story = str + array.join(" , ")
```

Comment:
Theres no need for a while loop to push an array's element to another. We can simply just say 
` let story = str + arr.join(" , ")` on line 267, it will do the same thing